### DISCUSSION TITLE
What are the *.browser.* module for?

### DISCUSSION THREAD

**Discussion by trusktr** [floating-ui/floating-ui#2840] [Q&A]
When to use `browser` files vs `.dom.esm.js` vs `.dom.mjs`? It isn't clear why there are so many variations of ES Modules:

- https://www.unpkg.com/browse/@floating-ui/dom@1.6.3/dist/floating-ui.dom.mjs
- https://www.unpkg.com/browse/@floating-ui/dom@1.6.3/dist/floating-ui.dom.esm.js
- https://www.unpkg.com/browse/@floating-ui/dom@1.6.3/dist/floating-ui.dom.browser.mjs
  - https://www.unpkg.com/browse/@floating-ui/dom@1.6.3/dist/floating-ui.dom.browser.min.mjs (this one is self explanatory if you're using `.browser.mjs` and then want to change to the minified version later)

**Comment by atomiks** [comment]
- `.mjs` is the modern file to be used in modern esm environments since the package doesn't have `type: "module"`.
- `.esm.js` is for webpack 4 and older bundlers.
- `.browser.mjs` no longer has a purpose since we removed the `__DEV__`/`process.env` validation checks, and is an alias for the regular `.mjs` file, kept for backwards-compatibility reasons. Edit: actually, I just remembered it has `@floating-ui/utils` bundled in as well, unlike `.mjs`. I think that was for backwards-compatibility reasons also.
- `.browser.min.mjs` is a minified file version of the above, I guess its purpose is that it's a pre-minified file for importing in browsers directly without a build step.
